code/modules/mob/living/simple_animal/abnormality/teth/crumbling_armor.dm

```diff
@@ -121,6 +132,80 @@
 			to_chat(user, "<span class='userdanger'>A strange power flows through you!</span>")
 	return
 
+/mob/living/simple_animal/hostile/abnormality/crumbling_armor/ZeroQliphoth(mob/living/carbon/human/user)
+	datum_reference.qliphoth_change(3) //no need for qliphoth to be stuck at 0
+	if(meltdown_cooldown > world.time)
+		return
+	meltdown_cooldown = world.time + meltdown_cooldown_time
+	MeltdownEffect()
+	return
+
+/mob/living/simple_animal/hostile/abnormality/crumbling_armor/proc/MeltdownEffect(mob/living/carbon/human/user)
+	var/list/potentialmarked = list()
+	var/list/marked = list()
+	var/mob/living/Y
+	for(var/mob/living/carbon/human/L in GLOB.player_list)
+		if(L.stat >= HARD_CRIT || L.sanity_lost || z != L.z) // Dead or in hard crit, insane, or on a different Z level.
+			continue
+		potentialmarked += L
+		to_chat(L, "<span class='userdanger'>You feel an overwhelming sense of dread.</span>")
+		playsound(get_turf(L), 'sound/abnormalities/crumbling/globalwarning.ogg', 25, FALSE, -1)
+
+	numbermarked = 1 + round(LAZYLEN(potentialmarked) / 5, 1) //1 + 1 in 5 potential players, to the nearest whole number
+	SLEEP_CHECK_DEATH(10 SECONDS)
+	for(var/i=numbermarked, i>=1, i--)
+		Y = pick(potentialmarked)
+		if(faction_check_mob(Y, FALSE) || Y.z != z || Y.stat == DEAD)
+			continue
+		if(Y in marked)
+			continue
+		marked+=Y
```

Comment:
The faction check should PROBABLY be done before this point. Instead of repeatedly picking and comparing to the other list, pick and remove the existing person from the list. 
```suggestion
		if(potentialmarked.len <= 0)
				break
		Y = pick(potentialmarked)
		potentialmarked -= Y
		if(faction_check_mob(Y, FALSE) || Y.z != z || Y.stat == DEAD)
			continue
		marked+=Y
```

Reply:
I believe I resolved this in my own way? It was a while ago, i'd appreciate if you took another look at it